src/features/slices/auth/authApiSlice.js

```diff
@@ -0,0 +1,28 @@
+import { apiSlice } from '../../api/apiSlice'
+import { loginUser } from './authSlice'
+
+export const authApiSlice = apiSlice.injectEndpoints({
+  endpoints: (builder) => ({
+    logIn: builder.mutation({
+      query: ({ credentials, endpoint }) => {
+        return {
+          url: `auth/${endpoint}/`,
+          method: 'POST',
+          body: { ...credentials },
+        }
+      },
+      async onQueryStarted(credentials, { dispatch, queryFulfilled }) {
+        try {
+          const { data } = await queryFulfilled
+          dispatch(loginUser({ ...data }))
+          localStorage.setItem('token', JSON.stringify(data?.access))
+          localStorage.setItem('refresh', JSON.stringify(data?.refresh))
+        } catch (error) {
+          console.log('APIslice error: ', error)
```

Comment:
now that we are consuming the api's, we shouldn't have `console.log` anymore. Please handle the error